src/colorizer/CanvasWithOverlay.ts

```diff
@@ -46,19 +70,51 @@ const defaultTimestampOptions: TimestampOptions = {
   visible: true,
 };
 
-const defaultBackgroundOptions: OverlayFillOptions = {
+const defaultOverlayBoxOptions: OverlayBoxOptions = {
   fill: "rgba(255, 255, 255, 0.8)",
   stroke: "rgba(0, 0, 0, 0.2)",
   paddingPx: new Vector2(10, 10),
   marginPx: new Vector2(12, 12),
   radiusPx: 4,
 };
 
+const defaultHeaderOptions: HeaderOptions = {
+  ...defaultStyleOptions,
+  fontSizePx: 16,
+  fill: "rgba(255, 255, 255, 1.0)",
+  stroke: "rgba(203, 203, 204, 1.0)",
+  paddingPx: new Vector2(10, 10),
+  visibleOnExport: false,
+};
+
+const defaultFooterOptions: FooterOptions = {
+  ...defaultHeaderOptions,
+  heightPx: 100,
+  paddingPx: new Vector2(10, 10),
+};
+
+const defaultLegendOptions: LegendOptions = {
+  ...defaultStyleOptions,
+  stroke: "rgba(203, 203, 204, 1.0)",
+  labelFontSizePx: 12,
+  labelFontColor: "black",
+
+  categoryPaddingPx: new Vector2(2, 2),
+  categoryLabelGapPx: 6,
+  categoryColGapPx: 32,
+  maxCategoricalWidthPx: 800,
+
+  maxColorRampWidthPx: 300,
+  rampRadiusPx: 4,
+};
+
+type RenderCallback = (origin: Vector2) => void;
+
 type RenderInfo = {
   /** Size of the element, in pixels. */
   sizePx: Vector2;
   /** Callback to render the element. */
-  render: () => void;
+  render: RenderCallback;
 };
```

Comment:
So this pattern is now used in all of the rendering logic inside `CanvasWithOverlay`. The way it works is that, for each of the rendered elements, there's a `getRenderer` method which will return a `RenderInfo` object. That contains the size of the element and a callback to actually render it, with an option to offset it by its origin (usually the top left corner).

This lets me do things like aligning elements, or sizing their containers based on the maximum dimensions of the content inside. The footer and overlay box both resize dynamically based on the visible content inside of them.